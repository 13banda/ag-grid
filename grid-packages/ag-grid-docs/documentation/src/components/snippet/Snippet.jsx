import React from 'react';
import Prism from 'prismjs';
import * as esprima from 'esprima';

export const Snippet = (props, framework) => {
    const suppliedSnippet = props.children[0].toString();

    // create syntax tree from supplied snippet
    const tree = esprima.parseScript(suppliedSnippet.trim(), {comment: true, range: true, loc: true});

    // convert syntax tree into a more convenient form, [{ propertyName: propertyAstObj }]
    let propertyMappings = parseTree(tree);

    // create FW specific snippet
    const snippet =
        framework === 'angular' ? createNgSnippet(propertyMappings) :
            framework === 'react' ? createReactSnippet(propertyMappings) :
                framework === 'vue' ? createVueSnippet(propertyMappings) :
                    createJsSnippet(propertyMappings);

    return <CodeSnippet code={snippet}/>;
};

function parseTree(tree) {
    const isVarDeclaration = node => node.type === 'VariableDeclaration' && Array.isArray(node.declarations);

    const commentsMap = tree.comments.reduce((acc, comment) => {
        acc[comment.loc.start.line] = comment.value;
        return acc;
    }, {});

    const parse = (acc, node) => {
        if (Array.isArray(node)) {
            node.forEach(n => parse(acc, n));
        } else if (isVarDeclaration(node)) {
            node.declarations.forEach(n => parse(acc, n));
        } else {
            const commentAbove = commentsMap[node.loc.start.line - 1];
            acc.push({name: node.key.name, comment: commentAbove, ...node.value});
        }
    }

    // simpler and faster to start here
    const root = tree.body[0].declarations[0].init.properties;

    // using array to preserve supplied order
    const propertyMappings = [];

    parse(propertyMappings, root);

    return propertyMappings;
}

const createJsSnippet = propertyMappings => {
    let res = 'const gridOptions = {';
    propertyMappings.forEach(prop => {
        if (prop.comment) {
            res += `\n\t//${prop.comment}`;
        }

        if (prop.name === 'columnDefs') {
            res += '\n\tcolumnDefs: [\n';
            res += formatColDefsAsObjs(prop.elements, 2).join('\n');
            res += '\n\t],';
        } else {
            res += `\n\t${prop.name}: ${prop.value},`;
        }
    });

    res += '\n\n\t// other grid options ...';
    res += '\n}';

    return res;
}

const createReactSnippet = propertyMappings => {
    let res = '<AgGridReact>';
    propertyMappings.forEach(prop => {
        if (prop.comment) {
            res += `\n\t//${prop.comment}`;
        }

        if (prop.name === 'columnDefs') {
            res += prop.elements.map(colDefAst => {
                const colDefObj = colDefAst.properties.map(property => {
                    return `${property.key.name}='${property.value.value}'`;
                });
                return `\n${pad(1)}<AgGridColumn ${colDefObj.join(', ')} />`;
            }).join('');
        }
    });

    res += '\n</AgGridReact>';

    return res;
}

const createNgSnippet = propertyMappings => {
    let res = `<ag-grid-angular
    [columnDefs]="columnDefs"
    // other grid options ...>
</ag-grid-angular>
\n`;

    return addPropsToNgOrVueSnippet(res, propertyMappings);
}

const createVueSnippet = propertyMappings => {
    let res = `<ag-grid-vue
    :columnDefs="columnDefs"
    // other grid options ...>
</ag-grid-vue>
\n`;

    return addPropsToNgOrVueSnippet(res, propertyMappings);
}

const addPropsToNgOrVueSnippet = (res, propertyMappings) => {
    for (let i = 0; i < propertyMappings.length; i++) {
        const prop = propertyMappings[i];
        if (prop.comment) {
            if (i > 0) res += `\n`;
            res += `//${prop.comment}\n`;
        }

        if (prop.name === 'columnDefs') {
            res += 'this.columnDefs = [\n';
            res += formatColDefsAsObjs(prop.elements, 1).join('\n');
            res += '\n];\n';
        } else {
            res += `this.${prop.name} = ${prop.value};`;
        }
    }

    return res;
}

const formatColDefsAsObjs = (colDefs, padding) => {
    return colDefs.map(colDefAst => {
        const colDefObj = colDefAst.properties.map(property => {
            return `${property.key.name}: '${property.value.value}'`;
        });
        return `${pad(padding)}{ ${colDefObj.join(', ')} },`;
    });
}

const pad = n => new Array(n).fill('\t').join('');

const CodeSnippet = ({code}) => <pre className="language-ts">
    <code dangerouslySetInnerHTML={{__html: Prism.highlight(code, Prism.languages.typescript, 'typescript')}}/>
</pre>;
