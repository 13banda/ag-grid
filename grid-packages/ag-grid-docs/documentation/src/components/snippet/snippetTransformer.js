import { parseScript } from 'esprima';
import { decorateWithComments } from "./commentDecorator";
import {
    createColDefSnippet,
    createReactColDefSnippet,
    getName,
    getReactValue,
    getValue,
    isArrayProperty,
    isLiteralProperty,
    isObjectProperty,
    tab,
} from "./snippetUtils";

export const transform = (snippet, framework, options) => {
    // create a syntax tree from the supplied snippet with comments associated to each node
    const tree = decorateWithComments(parseScript(snippet, {comment: true, loc: true}));

    return {
        'javascript': () => new JavascriptTransformer(options).transform(tree),
        'angular': () => new AngularTransformer(options).transform(tree),
        'react': () => new ReactTransformer(options).transform(tree),
        'vue': () => new VueTransformer(options).transform(tree),
    }[framework]();
}

// The SnippetTransformer is based around the 'Template Method' design pattern
class SnippetTransformer {
    initialDepth = 0;

    constructor(options) {
        this.options = options;
    }
    transform(tree) {
        return this.addFrameworkContext(this.parse(tree, this.initialDepth));
    }
    parse(tree, depth) {
        if (Array.isArray(tree)) {
            return tree.map(node => this.parse(node, depth + 1)).join('');

        } else if (isLiteralProperty(tree)) {
            return this.addComment(tree, depth) + this.parseLiteral(tree, depth);

        } else if (isArrayProperty(tree)) {
            return this.addComment(tree, depth) + this.parseArray(tree, depth);

        } else if (isObjectProperty(tree)) {
            return this.addComment(tree, depth) + this.parseObject(tree, depth);
        }
    }
}

class JavascriptTransformer extends SnippetTransformer {
    constructor(options) {
        super(options);
        if (this.options.suppressFrameworkContext) {
            this.initialDepth--;
        }
    }

    parseLiteral(property, depth) {
        return `${tab(depth)}${getName(property)}: ${getValue(property)},`;
    }
    parseArray(property, depth) {
        if (getName(property) === 'columnDefs') {
            let res = `${tab(depth)}${getName(property)}: [`;
            res += createColDefSnippet(property.value.elements, depth + 1);
            res += `,\n${tab(depth)}],`;
            return res;
        }
        console.error("Not yet implemented!");
    }
    parseObject(property, depth) {
        let res = `${tab(depth)}${getName(property)}: {`;
        res += property.value.properties.map(prop => this.parse(prop, depth + 1)).join('');
        res += `\n${tab(depth)}},`;
        return res;
    }
    addFrameworkContext(result) {
        if (this.options.suppressFrameworkContext) { return result.trim(); }
        return `const gridOptions = {${result}` +
               `\n\n${tab(1)}// other grid options here...` +
               '\n}';
    }
    addComment(property, depth) {
        return property.comment ? `\n${tab(depth)}//${property.comment}\n` : '\n';
    }
}

class AngularTransformer extends SnippetTransformer {
    // used when adding framework context
    propertiesVisited = [];

    parseLiteral(property, depth) {
        if (depth > 1) {
            // don't include nested object properties
            return `${tab(depth - 1)}${getName(property)}: ${getValue(property)},`;
        }
        this.propertiesVisited.push(getName(property));
        return `this.${getName(property)} = ${getValue(property)};`;
    }
    parseArray(property, depth) {
        this.propertiesVisited.push(getName(property));
        if (getName(property) === 'columnDefs') {
            return `this.${getName(property)}: [` + createColDefSnippet(property.value.elements, depth) + ',\n];';
        }
        console.error("Not yet implemented!");
    }
    parseObject(property, depth) {
        this.propertiesVisited.push(getName(property));
        const properties = property.value.properties.map(prop => this.parse(prop, depth + 1)).join('');
        return `this.${getName(property)}: {` + properties + '\n};';
    }
    addFrameworkContext(result) {
        if (this.options.suppressFrameworkContext) { return result.trim(); }
        const props = this.propertiesVisited.map(property => `${tab(1)}[${property}]="${property}"`).join('\n');
        return '<ag-grid-angular\n' + props +
            '\n    // other grid options ...>\n' +
            '</ag-grid-angular>\n' +
            result;
    }
    addComment(property, depth) {
        return property.comment ? `\n${tab(depth-1)}//${property.comment}\n` : '\n';
    }
}

class VueTransformer extends AngularTransformer {
    addFrameworkContext(result) {
        if (this.options.suppressFrameworkContext) { return result.trim(); }
        const props = this.propertiesVisited.map(property => `${tab(1)}[${property}]="${property}"`).join('\n');
        return '<ag-grid-vue\n' + props +
            '\n    // other grid options ...>\n' +
            '</ag-grid-vue>\n' +
            result;
    }
}

class ReactTransformer extends SnippetTransformer {
    propertySnippets = [];

    constructor(options) {
        super(options);
        if (this.options.suppressFrameworkContext) {
            this.initialDepth--;
        }
    }
    parseLiteral(property, depth) {
        this.propertySnippets.push(this.createLiteralSnippet(property, depth));
        return '';
    }
    parseArray(property, depth) {
        if (getName(property) === 'columnDefs') {
            let res = property.comment ? `\n${tab(1)}//${property.comment}` : '';
            res += createReactColDefSnippet(property.value.elements, depth);
            return res;
        }
        console.error("Not yet implemented!");
    }
    parseObject(property, depth) {
        let res = property.comment ? `\n${tab(depth)}//${property.comment}` : '';
        res += `\n${tab(depth)}${getName(property)}: {`;
        res += property.value.properties.map(prop => this.createLiteralSnippet(prop, depth + 1)).join(',');
        res += `\n${tab(depth)}},`;
        this.propertySnippets.push(res);
        return '';
    }
    addFrameworkContext(result) {
        if (this.propertySnippets.length > 0) {
            return `<AgGridReact` +
                   `${this.propertySnippets.join('')}` +
                   `\n${tab(1)}// other grid options ...\n>` +
                   `${result}\n</AgGridReact>`;
        }
        if (this.options.suppressFrameworkContext) {
            // framework context is only hidden if no properties exist for React
            return result.trim();
        }
        return `<AgGridReact>${result}\n</AgGridReact>`;
    }
    addComment() {
        return ''; // react comments added inplace
    }
    createLiteralSnippet(property, depth) {
        let res = property.comment ? `\n${tab(depth)}//${property.comment}` : '';
        return res + `\n${tab(depth)}${getName(property)}=${getReactValue(property)}`;
    }
}
