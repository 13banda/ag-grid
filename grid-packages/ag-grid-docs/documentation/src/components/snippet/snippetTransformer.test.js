import supportedFrameworks from "../../utils/supported-frameworks";
import { Snippet } from "./Snippet";
import React from "react";
import { transform } from './snippetTransformer';

// Utility method to verify snippets match saved snapshots for all frameworks and options
const runSnippetFrameworkTests = snippetToTest => {
    // with framework context (suppressFrameworkContext = false)
    it.each(supportedFrameworks)(`it should create '%s' snippets`, framework => {
        const generatedSnippet = transform(snippetToTest, framework, {suppressFrameworkContext: false});
        expect(generatedSnippet).toMatchSnapshot();
    });

    // without framework context (suppressFrameworkContext = true)
    it.each(supportedFrameworks)(`it should create '%s' snippets without framework context`, framework => {
        const generatedSnippet = transform(snippetToTest, framework, {suppressFrameworkContext: true});
        expect(generatedSnippet).toMatchSnapshot();
    });
}

// These tests are run for each framework!
describe('Snippet', () => {
    describe('given simple column definitions', () => {
        runSnippetFrameworkTests(
            `const gridOptions = {
                // define 3 columns
                columnDefs: [
                    { headerName: 'A', field: 'a' },
                    { headerName: 'B', field: 'b' },
                    { headerName: 'C', field: 'c' },
                ]
            }`
        );
    });
    describe('given column definitions with group columns', () => {
        runSnippetFrameworkTests(
            `const gridOptions = {
                // 2 levels of grouping
                columnDefs: [
                    {
                        headerName: 'G1',
                        children: [
                            { headerName: 'C1', field: 'c1' },
                            {
                                headerName: 'G2',
                                children: [
                                    { headerName: 'C2', field: 'c2' },
                                    { headerName: 'C3', field: 'c3' },
                                ],
                            },
                            { headerName: 'C4', field: 'c4' },
                        ]
                    },
                ]
            }`
        );
    });
    describe('given a mix of different properties', () => {
        runSnippetFrameworkTests(
            `const gridOptions = {
                // columnDefs property (special)
                columnDefs: [
                    { headerName: 'A', field: 'a' },
                    { headerName: 'B', field: 'b' },
                    { headerName: 'C', field: 'c' },
                ],
                // object property
                defaultColDef: {
                    // set every column width
                    width: 100
                },
                // numeric property
                rowHeight: 50,
                // boolean property
                rowDragManaged: true,
                // string property
                rowSelection: 'single',
            }`
        );
    });
});