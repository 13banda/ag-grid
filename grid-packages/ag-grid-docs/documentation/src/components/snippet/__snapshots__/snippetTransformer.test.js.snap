// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snippet given a mix of different properties it should create 'angular' snippets 1`] = `
"<ag-grid-angular
    [columnDefs]=\\"columnDefs\\"
    [defaultColDef]=\\"defaultColDef\\"
    [rowHeight]=\\"rowHeight\\"
    [rowDragManaged]=\\"rowDragManaged\\"
    [rowSelection]=\\"rowSelection\\"
    // other grid options ...>
</ag-grid-angular>

// columnDefs property (special)
this.columnDefs: [
    { headerName: 'A', field: 'a' },
    { headerName: 'B', field: 'b' },
    { headerName: 'C', field: 'c' },
];
// object property
this.defaultColDef: {
    // set every column width
    width: 100,
};
// numeric property
this.rowHeight = 50;
// boolean property
this.rowDragManaged = true;
// string property
this.rowSelection = 'single';"
`;

exports[`Snippet given a mix of different properties it should create 'angular' snippets without framework context 1`] = `
"// columnDefs property (special)
this.columnDefs: [
    { headerName: 'A', field: 'a' },
    { headerName: 'B', field: 'b' },
    { headerName: 'C', field: 'c' },
];
// object property
this.defaultColDef: {
    // set every column width
    width: 100,
};
// numeric property
this.rowHeight = 50;
// boolean property
this.rowDragManaged = true;
// string property
this.rowSelection = 'single';"
`;

exports[`Snippet given a mix of different properties it should create 'javascript' snippets 1`] = `
"const gridOptions = {
    // columnDefs property (special)
    columnDefs: [
        { headerName: 'A', field: 'a' },
        { headerName: 'B', field: 'b' },
        { headerName: 'C', field: 'c' },
    ],
    // object property
    defaultColDef: {
        // set every column width
        width: 100,
    },
    // numeric property
    rowHeight: 50,
    // boolean property
    rowDragManaged: true,
    // string property
    rowSelection: 'single',

    // other grid options here...
}"
`;

exports[`Snippet given a mix of different properties it should create 'javascript' snippets without framework context 1`] = `
"// columnDefs property (special)
columnDefs: [
    { headerName: 'A', field: 'a' },
    { headerName: 'B', field: 'b' },
    { headerName: 'C', field: 'c' },
],
// object property
defaultColDef: {
    // set every column width
    width: 100,
},
// numeric property
rowHeight: 50,
// boolean property
rowDragManaged: true,
// string property
rowSelection: 'single',"
`;

exports[`Snippet given a mix of different properties it should create 'react' snippets 1`] = `
"<AgGridReact
    // object property
    defaultColDef: {
        // set every column width
        width={100}
    },
    // numeric property
    rowHeight={50}
    // boolean property
    rowDragManaged={true}
    // string property
    rowSelection='single'
    // other grid options ...
>
    // columnDefs property (special)
    <AgGridColumn headerName='A' field='a' />
    <AgGridColumn headerName='B' field='b' />
    <AgGridColumn headerName='C' field='c' />
</AgGridReact>"
`;

exports[`Snippet given a mix of different properties it should create 'react' snippets without framework context 1`] = `
"<AgGridReact
// object property
defaultColDef: {
    // set every column width
    width={100}
},
// numeric property
rowHeight={50}
// boolean property
rowDragManaged={true}
// string property
rowSelection='single'
    // other grid options ...
>
    // columnDefs property (special)
<AgGridColumn headerName='A' field='a' />
<AgGridColumn headerName='B' field='b' />
<AgGridColumn headerName='C' field='c' />
</AgGridReact>"
`;

exports[`Snippet given a mix of different properties it should create 'vue' snippets 1`] = `
"<ag-grid-vue
    [columnDefs]=\\"columnDefs\\"
    [defaultColDef]=\\"defaultColDef\\"
    [rowHeight]=\\"rowHeight\\"
    [rowDragManaged]=\\"rowDragManaged\\"
    [rowSelection]=\\"rowSelection\\"
    // other grid options ...>
</ag-grid-vue>

// columnDefs property (special)
this.columnDefs: [
    { headerName: 'A', field: 'a' },
    { headerName: 'B', field: 'b' },
    { headerName: 'C', field: 'c' },
];
// object property
this.defaultColDef: {
    // set every column width
    width: 100,
};
// numeric property
this.rowHeight = 50;
// boolean property
this.rowDragManaged = true;
// string property
this.rowSelection = 'single';"
`;

exports[`Snippet given a mix of different properties it should create 'vue' snippets without framework context 1`] = `
"// columnDefs property (special)
this.columnDefs: [
    { headerName: 'A', field: 'a' },
    { headerName: 'B', field: 'b' },
    { headerName: 'C', field: 'c' },
];
// object property
this.defaultColDef: {
    // set every column width
    width: 100,
};
// numeric property
this.rowHeight = 50;
// boolean property
this.rowDragManaged = true;
// string property
this.rowSelection = 'single';"
`;

exports[`Snippet given column definitions with group columns it should create 'angular' snippets 1`] = `
"<ag-grid-angular
    [columnDefs]=\\"columnDefs\\"
    // other grid options ...>
</ag-grid-angular>

// 2 levels of grouping
this.columnDefs: [
    {
        headerName: 'G1',
        children: [
            { headerName: 'C1', field: 'c1' },
            {
                headerName: 'G2',
                children: [
                    { headerName: 'C2', field: 'c2' },
                    { headerName: 'C3', field: 'c3' },
                ],
            },
            { headerName: 'C4', field: 'c4' },
        ],
    },
];"
`;

exports[`Snippet given column definitions with group columns it should create 'angular' snippets without framework context 1`] = `
"// 2 levels of grouping
this.columnDefs: [
    {
        headerName: 'G1',
        children: [
            { headerName: 'C1', field: 'c1' },
            {
                headerName: 'G2',
                children: [
                    { headerName: 'C2', field: 'c2' },
                    { headerName: 'C3', field: 'c3' },
                ],
            },
            { headerName: 'C4', field: 'c4' },
        ],
    },
];"
`;

exports[`Snippet given column definitions with group columns it should create 'javascript' snippets 1`] = `
"const gridOptions = {
    // 2 levels of grouping
    columnDefs: [
        {
            headerName: 'G1',
            children: [
                { headerName: 'C1', field: 'c1' },
                {
                    headerName: 'G2',
                    children: [
                        { headerName: 'C2', field: 'c2' },
                        { headerName: 'C3', field: 'c3' },
                    ],
                },
                { headerName: 'C4', field: 'c4' },
            ],
        },
    ],

    // other grid options here...
}"
`;

exports[`Snippet given column definitions with group columns it should create 'javascript' snippets without framework context 1`] = `
"// 2 levels of grouping
columnDefs: [
    {
        headerName: 'G1',
        children: [
            { headerName: 'C1', field: 'c1' },
            {
                headerName: 'G2',
                children: [
                    { headerName: 'C2', field: 'c2' },
                    { headerName: 'C3', field: 'c3' },
                ],
            },
            { headerName: 'C4', field: 'c4' },
        ],
    },
],"
`;

exports[`Snippet given column definitions with group columns it should create 'react' snippets 1`] = `
"<AgGridReact>
    // 2 levels of grouping
    <AgGridColumn headerName='G1'>
        <AgGridColumn headerName='C1' field='c1' />
        <AgGridColumn headerName='G2'>
            <AgGridColumn headerName='C2' field='c2' />
            <AgGridColumn headerName='C3' field='c3' />
        </AgGridColumn>
        <AgGridColumn headerName='C4' field='c4' />
    </AgGridColumn>
</AgGridReact>"
`;

exports[`Snippet given column definitions with group columns it should create 'react' snippets without framework context 1`] = `
"// 2 levels of grouping
<AgGridColumn headerName='G1'>
    <AgGridColumn headerName='C1' field='c1' />
    <AgGridColumn headerName='G2'>
        <AgGridColumn headerName='C2' field='c2' />
        <AgGridColumn headerName='C3' field='c3' />
    </AgGridColumn>
    <AgGridColumn headerName='C4' field='c4' />
</AgGridColumn>"
`;

exports[`Snippet given column definitions with group columns it should create 'vue' snippets 1`] = `
"<ag-grid-vue
    [columnDefs]=\\"columnDefs\\"
    // other grid options ...>
</ag-grid-vue>

// 2 levels of grouping
this.columnDefs: [
    {
        headerName: 'G1',
        children: [
            { headerName: 'C1', field: 'c1' },
            {
                headerName: 'G2',
                children: [
                    { headerName: 'C2', field: 'c2' },
                    { headerName: 'C3', field: 'c3' },
                ],
            },
            { headerName: 'C4', field: 'c4' },
        ],
    },
];"
`;

exports[`Snippet given column definitions with group columns it should create 'vue' snippets without framework context 1`] = `
"// 2 levels of grouping
this.columnDefs: [
    {
        headerName: 'G1',
        children: [
            { headerName: 'C1', field: 'c1' },
            {
                headerName: 'G2',
                children: [
                    { headerName: 'C2', field: 'c2' },
                    { headerName: 'C3', field: 'c3' },
                ],
            },
            { headerName: 'C4', field: 'c4' },
        ],
    },
];"
`;

exports[`Snippet given simple column definitions it should create 'angular' snippets 1`] = `
"<ag-grid-angular
    [columnDefs]=\\"columnDefs\\"
    // other grid options ...>
</ag-grid-angular>

// define 3 columns
this.columnDefs: [
    { headerName: 'A', field: 'a' },
    { headerName: 'B', field: 'b' },
    { headerName: 'C', field: 'c' },
];"
`;

exports[`Snippet given simple column definitions it should create 'angular' snippets without framework context 1`] = `
"// define 3 columns
this.columnDefs: [
    { headerName: 'A', field: 'a' },
    { headerName: 'B', field: 'b' },
    { headerName: 'C', field: 'c' },
];"
`;

exports[`Snippet given simple column definitions it should create 'javascript' snippets 1`] = `
"const gridOptions = {
    // define 3 columns
    columnDefs: [
        { headerName: 'A', field: 'a' },
        { headerName: 'B', field: 'b' },
        { headerName: 'C', field: 'c' },
    ],

    // other grid options here...
}"
`;

exports[`Snippet given simple column definitions it should create 'javascript' snippets without framework context 1`] = `
"// define 3 columns
columnDefs: [
    { headerName: 'A', field: 'a' },
    { headerName: 'B', field: 'b' },
    { headerName: 'C', field: 'c' },
],"
`;

exports[`Snippet given simple column definitions it should create 'react' snippets 1`] = `
"<AgGridReact>
    // define 3 columns
    <AgGridColumn headerName='A' field='a' />
    <AgGridColumn headerName='B' field='b' />
    <AgGridColumn headerName='C' field='c' />
</AgGridReact>"
`;

exports[`Snippet given simple column definitions it should create 'react' snippets without framework context 1`] = `
"// define 3 columns
<AgGridColumn headerName='A' field='a' />
<AgGridColumn headerName='B' field='b' />
<AgGridColumn headerName='C' field='c' />"
`;

exports[`Snippet given simple column definitions it should create 'vue' snippets 1`] = `
"<ag-grid-vue
    [columnDefs]=\\"columnDefs\\"
    // other grid options ...>
</ag-grid-vue>

// define 3 columns
this.columnDefs: [
    { headerName: 'A', field: 'a' },
    { headerName: 'B', field: 'b' },
    { headerName: 'C', field: 'c' },
];"
`;

exports[`Snippet given simple column definitions it should create 'vue' snippets without framework context 1`] = `
"// define 3 columns
this.columnDefs: [
    { headerName: 'A', field: 'a' },
    { headerName: 'B', field: 'b' },
    { headerName: 'C', field: 'c' },
];"
`;
